packages/analytics/src/helpers.ts

```diff
@@ -166,6 +173,12 @@ async function gtagOnEvent(
     // Workaround for http://b/141370449 - third argument cannot be undefined.
     gtagCore(GtagCommand.EVENT, measurementId, gtagParams || {});
```

Comment:
It is out of scope of this PR, but we should move the `gtagCore` call outside of the try catch block. It should execute regardless if our initialization failed or not. The current implementation will prevent people from sending events to a non-firebase managed measurementId if any of our initialization failed.

Reply:
Another problem is we can not use `Promise.all` to wait for all promises, because it will reject if any of the promises was rejected. (Sorry I didn't catch it before!). 

Instead, we should use `Promise.allSettled` which is part of es2020 which is too new to use directly. There is an easy polyfill we can do before putting the promises into `initializationPromisesMap` and `dynamicConfigPromisesList ` (and other promise maps where we do `Promise.all`), like so:

```
function reflect(promise){
    return promise.then(function(v){ return {v:v, status: "fulfilled" }},
                        function(e){ return {e:e, status: "rejected" }});
}

initializationPromisesMap[appId] = reflect(initializeIds(\*...*\));

```

Reply:
We should resolve the above issues in a separate PR.